src/stores/DetailsForm/Store.ts

```diff
@@ -213,6 +213,14 @@ export const getInitialState = (
                     setHydrationErrorsExist(true);
                 }
             } else {
+                const { setDetails_connector } = get();
+                setDetails_connector({
+                    id: connectorId,
+                    iconPath: '',
+                    imageName: '',
+                    imagePath: '',
+                    connectorId,
+                });
```

Comment:
This is a hacky solution I introduced so I can hydrate just the `connectorId` without having to fetch anything else about the connector. The idea is when we want to test a new connector, we create a row for it in the `connectors` table, but no `connector_tags` row. We specify `oauth2_client_id` and `oauth2_client_secret` in `connectors` row and that is used to drive the OAuth flow.

I'm open to updating this logic to be perhaps more specific for the TestJsonForms page instead of being in this `else` block, or perhaps somehow moving it somewhere else so it doesn't interfere with the general logic at all. Let me know what you think.

Reply:
We should probably move this given we really want to reduce potential impact that the text page has on the application. 

I will need to think about this a bit.

Reply:
@travjenkins I introduced a new Workflow value `test_json_forms` and added an if condition in the `hydrateState` to only apply this logic if `workflow == 'test_json_forms'`, do you think this is sufficient isolation of this logic?

Reply:
Yeah - that is enough. I was thinking of that as one of the possible options and that seems easiest. Also makes it clear for future work that this store is used in a test page.